rfcs/0000-kuadrant-release-process.md

```diff
@@ -0,0 +1,205 @@
+what's happening and what to expect. The Kuadrant suite is composed of several components, each of them with its own
+set of artifacts and versioning scheme. Defining the release process of the whole suite is a complex task, and it's
+not only about the technical details of releasing the components, but also about the communication and transparency
+with the community, the definition of the frequency of the releases, and when it's ready to be handover to other teams like
+QA. This section aims to provide guidelines for the different aspects of the release process.
+
+## Components and versioning
+
+The set of components that are part of the _Kuadrant suite_ are the following:
+
+- [Authorino](https://github.com/Kuadrant/authorino): Kubernetes-native authorization service for tailor-made Zero Trust API security.
+- [Authorino Operator](https://github.com/Kuadrant/authorino-operator): A Kubernetes Operator to manage Authorino instances.
+- [Limitador](https://github.com/Kuadrant/limitador): A generic rate-limiter written in Rust.
+- [Limitador Operator](https://github.com/Kuadrant/limitador-operator/): A Kubernetes Operator to manage Limitador deployments.
+- [Wasm Shim](https://github.com/Kuadrant/wasm-shim/): A Proxy-Wasm module written in Rust, acting as a shim between Envoy and Limitador.
+- [Multicluster Gateway Controller](https://github.com/Kuadrant/multicluster-gateway-controller): Provides multi-cluster
+connectivity and global load balancing.
+- [Kuadrant Operator](https://github.com/Kuadrant/kuadrant-operator/): The Operator to install and manage the lifecycle
+  of the Kuadrant components deployments.
+
+Each of them needs to be versioned independently, and the versioning scheme should follow [Semantic Versioning]. At
+the time of cutting a release for any of them, it's important to keep in mind what section of the version to bump,
+given a version number MAJOR.MINOR.PATCH, increment the:
+
+* MAJOR version when you make incompatible API changes
+* MINOR version when you add functionality in a backward compatible manner
+* PATCH version when you make backward compatible bug fixes
+
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+A more detailed explanation of the versioning scheme can be found in the [Semantic Versioning](https://semver.org/) website.
+
+By releasing a new version of Kuadrant, we mean releasing the set of components with their corresponding semantic versioning,
+some of them maybe freshly released, or others still using versioning from the previous one, and being the version of the
+**Kuadrant Operator** the one that defines the version of the whole suite.
+
+The technical details of how to release each component are out of the scope of this RFC and could be found in the
+[Kuadrant components CI/CD](https://github.com/Kuadrant/architecture/pull/41) RFC.
+
+## Handover to QA
```

Comment:
I keep thinking that this should not be part of the release process upstream. We should make sure the test suite is always as up to date as possible and those tests that exist should pass. But that's it. Any gap that exists between features and QE coverage should be minimized by optimizing our development process, not by adding a post development QE process. By adding such a post development QE step we've already thrown in the towel.

Reply:
But the reason for having a small QE part on the release is not about testsuite, I believe there are tasks which are simply unfeasible to run frequently (e.g. huge overhead) and hence are ideal for the release testing, for example, running on all supported configurations (single + multicluster).

Some tasks can only be done after we have a final bundle completed, like testing quickstart, verifying OLM deliverables, and verifying that all images are correct, which cannot be done beforehand.

Reply:
Ok, maybe the term "Handover to QE" is what triggered my response as it sounds heavy and slow to me. Also, do we "support configurations"? I'd say we "enable configurations". And if there are issues they can be fixed. I guess what I'm saying is that we should not drop the full weight of downstream on upstream. That doesn't mean we should ship sloppy software but it does mean we can make a different risk calculation.

Reply:
I agree with both making a better name for the step and that we should not drop the full weight downstream on upstream. I believe the only thing that have not yet reached an agreement on is the risk calculation itself.

While it would be completely ok to ship without QE step at release at this stage. However, I believe the value of additional QE testing on release outweighs the delay it causes (e.g. the 2-3 days maximum that we commit ourselves to do). We are also definitely not treating it as a downstream release, we do have different requirements for upstream (only automation run, fixed time to test vs test everything etc).

I do not see the inclusion of the "QE" step in a release process as a downstream thing, but as a last sanity check that everything is ok and as a place to do testing on the all assembled pieces compared to individual components (Note: This argument does not hold for individual component releases, however, the still the last run of the automation on a final build is still a good sanity check). There is a fine balance of testing required that we need to find and what exactly should be tested for upstream but I do not think that eliminates the step, only reduces its length.

Regarding the `support configuration` vs `enable configurations`, enable is a better name for it for sure (I keep defaulting to downstream language because I have no idea how to call it otherwise), however, I think it is also an important part of the testing for a project where multicluster is a key feature, so testing it on dedicated multiple production-grade clusters makes sense, just not on every PR

Reply:
This is a helpful conversation and I think we're pretty close to agreeing. How about we call the step "QE sanity check"? We might also add that we aim for this step to take 0 extra time and to be fully automated at some moment in the future but that for now we agree on an expected duration of 1-2 days and a maximum duration of 3 days.

Reply:
I am fully onboard with that idea! Thanks for suggesting it!